system/ee/ExpressionEngine/Addons/query/mod.query.php

```diff
@@ -50,15 +50,40 @@ public function __construct()
             $results = array_slice($results, $pagination->offset, $pagination->per_page);
         }
 
-        $str = ee()->TMPL->parse_variables(ee()->TMPL->tagdata, array_values($results));
+        $parsed = '';
 
-        if (ee()->TMPL->fetch_param('parse_files') === 'yes' && strpos($str, '{filedir_') !== false) {
+        if (get_bool_from_string(ee()->TMPL->fetch_param('parse_bases', config_item('parse_variables_query_results_by_default')))) {
+            $count = 0;
+
+            foreach ($results as $row) {
+                $row['count'] = ++$count;
+                $row['total_results'] = $query->num_rows;
+
+                $chunk = ee()->TMPL->parse_variables_row(ee()->TMPL->tagdata, $row);
+
+                $chunk = ee()->TMPL->parse_switch($chunk, $count - 1);
+
+                if (strpos($chunk, LD . 'base_') !== false) {
+                    $chunk = isset($row['site_id'])
+                        ? parse_config_variables(
+                            $chunk,
+                            ee()->config->get_cached_site_prefs($row['site_id'])
+                        )
+                        : parse_config_variables($chunk);
+                }
+                $parsed .= $chunk;
+            }
+        } else {
+            $parsed = ee()->TMPL->parse_variables(ee()->TMPL->tagdata, $results);
+        }
+
+        if (get_bool_from_string(ee()->TMPL->fetch_param('parse_files', config_item('parse_variables_query_results_by_default'))) && strpos($parsed, LD . 'filedir_') !== false) {
```

Comment:
Shall we not get same (or better) result if we just apply `ee()->TMPL->parse_globals($str)` at the end?

Reply:
@intoeetive , I can't see the point. [`parse_globals()`](https://github.com/ExpressionEngine/ExpressionEngine/blob/1b8071562a910323032f6a006a7973ddd63f6a9a/system/ee/legacy/libraries/Template.php#L3012) looks a total overkill for this case. It looks for a lot of things we can't get from database, except in case we query for templates themselves, which could create a total mess.

Even more, how would we get data from different sites? I'm using a column `site_id` to check for the right value.